### ISSUE TITLE
[FEAT]: Difficulty-based Train-test split of Sotopia data

### ISSUE THREAD

**Issue by ruiyiw** [sotopia-lab/sotopia-pi#23]
### Description

The current split of fine-tuning data is based on task difficulty/variance:

All GPT-4/GPT-4 dialogues (with tag `"gpt-4_gpt-4_v0.0.1_clean"`): 90 * 5 = 450

Train set: 76 (easy scenarios) * 5 = 380
Test set: 14 (hard scenarios) * 5 = 70

_Choosing hard scenarios based on_ `EnvironmentList.find(EnvironmentList.name == "hard_env_set").all()`

**An agreement on this archived issue should resolve issues #20 #5 #6**
